src/main/java/org/ladysnake/blabber/api/DialogueIllustration.java

```diff
@@ -0,0 +1,46 @@
+package org.ladysnake.blabber.api;
+
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.entity.Entity;
+import net.minecraft.network.PacketByteBuf;
+import net.minecraft.server.command.ServerCommandSource;
+import org.jetbrains.annotations.Nullable;
+
+public interface DialogueIllustration {
+    /**
+     * Draw this illustration to the screen.
+     *
+     * @param context a context to draw in
+     * @param textRenderer a text renderer
+     * @param x the x position it should be drawn relative to
+     * @param y the y position it should be drawn relative to
+     * @param mouseX the current x mouse position
+     * @param mouseY the current y mouse position
+     * @param tickDelta how much time has passed since last frame
+     */
+    void render(DrawContext context, TextRenderer textRenderer, int x, int y, int mouseX, int mouseY, float tickDelta);
+
+    /**
+     * Write the data this illustration needs to be drawn client-side to a packet
+     * @param packetByteBuf the packet to write to
+     */
+    void writeToPacket(PacketByteBuf packetByteBuf);
```

Comment:
Not really, actually, because that would make it

```
public abstract void writeToPacket(PacketByteBuf buf, T toWrite);
```

which makes it impossible (as far as I can think of) to make a `DialogueIllustration i` write itself; `i.getType()` loses type information, so `writeToPacket` would need to take in an unknown type that we don't have. We could use some hacky trickery by making the IllustrationType cast(??) but that's cursed as hell.

Reply:
Ah yes, fair point... Looks like Mojang likes unchecked casts in those instances though, so I say we follow that